azkaban-web-server/src/main/bash/azkaban-web-shutdown.sh

```diff
@@ -2,14 +2,33 @@
 # Shutdown script for azkaban web server
 installdir="$(dirname $0)/.."
 
+#currdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+#installdir=$(dirname "${currdir}")
+maxtry=5
 pid=`cat ${installdir}/currentpid`
-echo "Killing Web Server. [pid: $pid]"
 
-kill ${pid}
-if [ $? -ne 0 ]; then
-    echo "Error: Shutdown failed"
-    exit 1;
+if [ -z $pid ]; then
+  echo "currentpid file doesn't exist in ${installdir}, shutdown completed"
+  exit 0
 fi
 
-rm  ${installdir}/currentpid
-echo "done."
+for try in $(seq 1 $maxtry); do
+  if [ ! -z $pid ]; then
+    echo "Killing Web Server. [pid: $pid], $try th try"
+    kill ${pid}
+    if [ -n "$(ps -p $pid -o pid=)" ]; then
+      echo "web server is not dead [pid: $pid]"
+      if [ $try -lt $maxtry ]; then
+        echo "sleeping for a few seconds before retry"
+        sleep 10 
+      fi
+    else 
+      rm  ${installdir}/currentpid
+      echo "shutdown succeeded"
+      exit 0
+    fi
+  fi
+done
+
+echo "Error: Shutdown failed"
+exit 1
```

Comment:
so web server shutdown fails here, deployment should also fail since starting another instance of web server is risky. But if the deployment fails because of this, we still have to manually kill the web server by doing kill -9. So I think we should have kill -9 as the final fallback in the script.